mixnet/protocol/src/lib.rs

```diff
@@ -1,25 +1,20 @@
-use sphinx_packet::SphinxPacket;
+use sphinx_packet::{payload::Payload, SphinxPacket};
 use std::error::Error;
-use std::io::Cursor;
 
 use tokio::io::{AsyncRead, AsyncReadExt, AsyncWrite, AsyncWriteExt};
 
-pub enum Body<'a> {
+pub enum Body {
     SphinxPacket(Box<SphinxPacket>),
-    FinalPayload(Box<dyn AsyncRead + Unpin + Send + 'a>),
+    FinalPayload(Payload),
```

Comment:
@danielSanchezQ Now, we need to forward this body via `mpsc::channel` to the `ClientNotifier`, which is added in this PR to send payloads to the client connected. However, I couldn't find a way to send this `AsyncRead` reference via channel. I'm probably not familiar with reference lifetime, but I think it's okay to go with the simplest apporoach for now because we anyway need to introduce reusing connections between mixnodes in near future.

Reply:
Well, you can send the `Box<dyn AsyncRead + ...>`  to the channel. But I do not think it is what you want here. For me it makes for sense to have this `Payload`.